dtdl-parser/src/field_info_impl.rs

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+use serde_json::Value;
+use std::any::Any;
+use std::collections::HashMap;
+
+use crate::dtmi::Dtmi;
+use crate::entity_info::EntityInfo;
+use crate::entity_kind::EntityKind;
+use crate::field_info::FieldInfo;
+use crate::named_entity_info::NamedEntityInfo;
+use crate::schema_field_info::SchemaFieldInfo;
+use crate::schema_info::SchemaInfo;
+
+pub struct FieldInfoImpl {
+    // EntityInfo
```

Comment:
EntityInfo seems to repeat for structs in this PR. Why not use struct composition and have EntityInfo struct be a property of the final struct.

Or if you must have flat properties on the final struct, why not use a macro?

Any of the above will reduce the amount of implementation and tests.

Reply:
William suggested using macros to remove the repetition.

See https://stackoverflow.com/questions/39150216/implementing-a-trait-for-multiple-types-at-once

I'm still deciding whether I should fix this in this PR or a separate one.

Reply:
Macros is an option, if composition does not fit. Does it not?

Reply:
I can use either.

Reply:
IMHO, composition is more straightforward unless flat struct is desired.

Consider having an issue to track that if it's going to be outside of this PR, or consider having this comment unresolved if the change will be part of this PR.